core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksInodeStore.java

```diff
@@ -493,14 +538,29 @@ public Set<MutableInode<?>> allInodes() {
   }
 
   /**
-   * The name is intentional, in order to distinguish from the {@code Iterable} interface.
+   * Acquires an iterator to iterate all Inodes in RocksDB.
+   * A shared lock will be acquired when this iterator is created, and released when:
+   * 1. This iterator is complete.
+   * 2. At each step, the iterator finds the RocksDB is closing and aborts voluntarily.
+   *
+   * Except tests, this iterator is only used in:
+   * 1. {@link alluxio.master.journal.tool.AbstractJournalDumper} which translates RocksDB
+   *    checkpoints to a human-readable form.
    *
    * @return an iterator over stored inodes
    */
   public CloseableIterator<InodeView> getCloseableIterator() {
-    return RocksUtils.createCloseableIterator(
-        db().newIterator(mInodesColumn.get(), mIteratorOption),
-        (iter) -> getMutable(Longs.fromByteArray(iter.key()), ReadOption.defaults()).get());
+    try (RocksSharedLockHandle lock = mRocksStore.checkAndAcquireSharedLock()) {
+      RocksSharedLockHandle readLock = mRocksStore.checkAndAcquireSharedLock();
+      return RocksUtils.createCloseableIterator(
+          db().newIterator(mInodesColumn.get(), mIteratorOption),
+          (iter) -> getMutable(Longs.fromByteArray(iter.key()), ReadOption.defaults()).get(),
+          // TODO(jiacheng): validate an ex is thrown in UT
+          () -> {
+            mRocksStore.shouldAbort(lock.mDbVersion);
```

Comment:
Or here, could it be possible to move the abort checks to inside the next() and hasNext() calls of the iterator created by the call db().newIterator() in RocksStore?

Reply:
Same as https://github.com/Alluxio/alluxio/pull/17171#discussion_r1169062360 we don't have control over `db().newIterator()` and we can only wrap around that.